### PR TITLE
[infrt] add infrt dialect ir.

### PR DESCRIPTION
**Pull request by winter-wang**
### PR types

New features
### PR changes

Others
### Describe

添加infrt dialect ir

### CODE REVIEW

File: paddle/infrt/dialect/infrt/infrt_ops.td [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#define Infrt_OpS
+
+include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+def Infrt_Dialect : Dialect {
+  let summary =
+    "A dialect containing the Infrt Attributes, Operations, and Types";
+
+  let name = "Infrt";
```
Comment (Superjomn) [inline paddle/infrt/dialect/infrt/infrt_ops.td:11]: infrt ? 

It is better to replace Infrt_Dialect with  INFRT_Dialect, to keep consistent with the other places.

File: paddle/infrt/dialect/infrt/infrt_ops.td [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+// Type definitions
+
+// Base class for Infrt dialect types.
+class Infrt_Type<string name, list<Trait> traits = [],
+                   string baseCppClass = "::mlir::Type">
+    : TypeDef<Infrt_Dialect, name, traits, baseCppClass> {
+}
+
+def LoDTensor : Infrt_Type<"LoDTensor"> {
```
Comment (Superjomn) [inline paddle/infrt/dialect/infrt/infrt_ops.td:23]: Currently the LoDTensor is just a DenseTensor without LOD operations?
Is it necessary to expose such concept to IR ? @Shixiaowei02

File: paddle/infrt/dialect/infrt/infrt_ops.td (hunk 1 above)
Comment (winter-wang) [reply paddle/infrt/dialect/infrt/infrt_ops.td:11]: ok

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**